File: fixtures/scale/java/large/src/cryptobench/ciphers/rot13/Rot13Suite.java
package cryptobench.ciphers.rot13;

import cryptobench.core.Cipher;
import cryptobench.core.CipherSuite;
import cryptobench.core.SuiteResult;
import cryptobench.verify.RoundTrip;

public class Rot13Suite implements CipherSuite {
    @Override
    public String name() {
        return "rot13";
    }

    @Override
    public SuiteResult run() {
        Cipher cipher = new Rot13Cipher(Rot13Key.defaultKey());
        int passed = 0;
        int failed = 0;
        long start = System.nanoTime();
        for (String sample : Rot13Vectors.samples()) {
            if (RoundTrip.check(cipher, sample)) passed++; else failed++;
        }
        return new SuiteResult(name(), passed, failed, System.nanoTime() - start);
    }
}

File: fixtures/scale/kotlin/large/src/cryptobench/ciphers/rot13/Rot13Suite.kt
package cryptobench.ciphers.rot13

import cryptobench.core.Cipher
import cryptobench.core.CipherSuite
import cryptobench.core.SuiteResult
import cryptobench.verify.roundTrip

class Rot13Suite : CipherSuite {
    override fun name(): String = "rot13"

    override fun run(): SuiteResult {
        val cipher: Cipher = Rot13Cipher(Rot13Key.default())
        var passed = 0
        var failed = 0
        val start = System.nanoTime()
        for (sample in Rot13Vectors.samples()) {
            if (roundTrip(cipher, sample)) passed++ else failed++
        }
        return SuiteResult(name(), passed, failed, System.nanoTime() - start)
    }
}
